homeassistant/components/zha/helpers.py

```diff
@@ -82,3 +65,23 @@ def get_discovery_info(hass, discovery_info):
             "%s: failed to set reporting for '%s' attr on '%s' cluster: %s",
             entity_id, attr_name, cluster_name, str(ex)
         )
+
+
+async def check_zigpy_connection(usb_path, radio_type):
+    """Test zigpy radio connection."""
+    if radio_type == RadioType.ezsp.name:
+        import bellows.ezsp
+        from bellows.zigbee.application import ControllerApplication
+        radio = bellows.ezsp.EZSP()
+    elif radio_type == RadioType.xbee.name:
+        import zigpy_xbee.api
+        from zigpy_xbee.zigbee.application import ControllerApplication
+        radio = zigpy_xbee.api.XBee()
+    try:
+        await radio.connect(usb_path, DEFAULT_BAUDRATE)
+        controller = ControllerApplication(radio, DEFAULT_DATABASE_PATH)
+        await controller.startup(auto_form=True)
+        radio.close()
+    except Exception:  # pylint: disable=broad-except
```

Comment:
Don't we know what error to expect here?

Reply:
I guess many things can go wrong here. Wrong port, permissions, zigpy driver, mode, etc.